### PR TITLE
Add support for checking availability of organization names

### LINKED ISSUE #14: Support for organization
NPM has organization which work along with the scoped package features. Therefore, I think it would be handy to do:

```shell
npm-name @elemental
```

The `@` to detect an org instead of a package name.

### PR DESCRIPTION
**Pull request by sidoshi**
Fixes #14

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,8 +9,16 @@ const validate = require('validate-npm-package-name');
 
 class InvalidNameError extends Error {}
 
+const orgRegex = /^@[a-z\d][\w-.]+\/?$/;
+
 const request = async (name, options) => {
 	const registryUrl = normalizeUrl(options.registryUrl || configuredRegistryUrl);
+	const npmOrgUrl = 'https://www.npmjs.com/org/';
```
Comment (sindresorhus) [inline index.js]: This is a constant and can be placed at the top-level.

File: index.js [hunk 2]
```diff
@@ -9,8 +9,16 @@ const validate = require('validate-npm-package-name');
...
+
+	const isOrg = orgRegex.test(name);
```
Comment (sindresorhus) [inline index.js]: You can just do `name.startsWith('@')`.

And:
```suggestion
	const isOrganization = orgRegex.test(name);
```

**Comment by sindresorhus** [comment]
You need to document the support in the CLI and the readme.

**Comment by sindresorhus** [comment]
This only handles public orgs though, so it's not that useful. Could you give this a shot:

> Could also potentially do the same form request as done here: npmjs.com/org/create (Try creating an org named ava and you'll see they notify that it already exists.

**Comment by sidoshi** [comment]
> This only handles public orgs though, so it's not that useful. Could you give this a shot:

The form request done there is a post request on `npmjs.com/org/create` which requires handling authentication and CSRF tokens. Also if the name is available, with right credentials the request would create a new organization. I couldn't find a way to only check if the org is available without actually creating the org.

**Comment by sidoshi** [comment]
Unfortunately, I don't have any private orgs to experiment with. But I think it would be possible that we would be getting enough information from the response to HEAD `npmjs.com/org/create/<private-org>` request to support private orgs as well.

File: index.js (hunk 2 above)
Comment (sidoshi) [reply index.js]: `name.startsWith('@')` would be true even for scoped packages like `@ava/init`

**Comment by sidoshi** [comment]
> This only handles public orgs though, so it's not that useful.

I did some research and I don't think we can't create private organizations in NPM. There are only private packages AFAICT and the organization with only private packages would simply show 0 packages to users without access. 

So the current implementation would work as expected. Though, I am not sure about this. Do you have any thoughts on this?

**Comment by sindresorhus** [comment]
Alright. Looks good 👌

**Comment by sindresorhus** [comment]
It's out: https://github.com/sindresorhus/npm-name/releases/tag/v5.5.0

Would you be able to update https://github.com/sindresorhus/npm-name-cli too? 🙏

Also note my additional changes in https://github.com/sindresorhus/npm-name/commit/081c970f7c1b03fbd6c827312a1218b8ceba0587

**Comment by sidoshi** [comment]
Sure, I will update the cli and submit a PR soon.

**Pull request merged**